### DISCUSSION TITLE
is there a neat way to reduce entire dataset dimension size x and y without changing datatype for variables?

### DISCUSSION THREAD

**Discussion by oswald1234** [pydata/xarray#6465] [Q&A] [answered]
I'm trying to reduce my xarray dataset dimension (x and y) by one, e.g. 257x257 to 256x256.

This code is what I have tried:

`
if cube.dims['x'] > patch_size:
     cube=cube.where((cube.y<cube.y.data.max()) & (cube.x<cube.x.max()),drop=True)
`

When I run this code, all variables change the data type to float64 (probably because where() statement converts to "nan" while selecting data). Is there a better way of doing this without changing the data type of all the variables to float64?

**Comment by keewis** [comment, marked as answer]
this really depends on what value you want to drop, but you could use `isel` with the appropriate `slice`, e.g. to drop the last element:
```python
cube.isel(x=slice(None, -1), y=slice(None, -1))
```
